### ISSUE TITLE
Return Url looks incorrect

### ISSUE THREAD

**Issue by Sathiyanathan-Macro** [IdentityServer/IdentityServer4#3012]
Hi,

My Auth Server is https://xyz.net/oauth

We are working on ASPNETCore Web API project

I can able to get the access token by calling the Endpoint https://xyz.net/oauth/connect/token with Client Id and Secret.

While calling Authorize Endpoint : the Login page is connected without any issue. But after successful login and consent then i call redirect function with returnurl parameter.

But returnurl showing https://xyz.net/connect/authorize/callbackurl... and browser showing below error message

**The resource you are looking for has been removed, had its name changed, or is temporarily unavailable.**

I think IdentityServer missed out sub directory oauth

When insert oauth between xyz.net/ and connect then authorization code successfully sent to the redirect Uri 

We are using Azure server with sub directory for different application 

Please help us

**Comment by Sathiyanathan-Macro** [comment]
Hi,
Have you got any solution on this issue?

**Comment by Sathiyanathan-Macro** [comment]
Hi,

Can you please respond this issues asap. Bit Urgent....

**Comment by brockallen** [comment]
Can you tell us how you're setting the base path `~/oauth`?

**Comment by Sathiyanathan-Macro** [comment]
No, We are not set any base path.

We have added the Virtual Directory in Azure Environment.

**Comment by leastprivilege** [comment]
That is probably the issue, you set a "vdir" outside of ASP.NET Core - so it doesn't anything about it.

**Comment by brockallen** [comment]
I don't have a sample in azure, but I tried to repo this locally by setting up my IdentityServer in IIS under an application not at the root (I chose `~/path` as my child app name):

![image](https://user-images.githubusercontent.com/649492/56867311-b8ad3e00-69b1-11e9-9149-c0c10d3a83be.png)

I then connected and ran my app:

![image](https://user-images.githubusercontent.com/649492/56867319-d4b0df80-69b1-11e9-8479-040c5a19c10c.png)

it might be hard to tell, but all of the redirects (login, consent, and back to the various authorize callback paths) are all working properly. 

In short, I can't repo your issue without more investigation from your side. Sorry.

**Comment by Sathiyanathan-Macro** [comment]
Brockallen,

I'll check my side as well and give you some test cases to you for more understanding.
